src/frontend/pages/Signup/Signup.jsx

```diff
@@ -0,0 +1,179 @@
+import { Link, useNavigate } from "react-router-dom";
+import "../Signin/Signin.css";
+import "../../../App.css";
+import { useState } from "react";
+import { useAuth } from "../../context";
+import { signupService } from "../../services";
+const Signup = () => {
+  const { setAuth } = useAuth();
+  const navigate = useNavigate();
+  const [user, setUser] = useState({
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+  });
+
+  const [showPassword, setShowPassword] = useState(false);
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
+  const signupFormSubmitHandler = async user => {
+    const [name, authToken] = await signupService(user);
+    if (authToken !== undefined) {
+      localStorage.setItem("AUTH_TOKEN", authToken);
+      localStorage.setItem("username", name);
+      setAuth(auth => ({
+        ...auth,
+        status: true,
+        token: localStorage.getItem("AUTH_TOKEN"),
+        username: name,
+      }));
+      navigate("/");
+    }
+  };
+  return (
+    <section
+      id="signup-card-wrapper"
+      className="auth-card-wrapper container-center"
+    >
+      <form
+        className="auth-card rounded-md box-shadow-sm my-8"
+        onSubmit={e => {
+          e.preventDefault();
+          signupFormSubmitHandler(user);
+        }}
+      >
+        <h1 className="headline-lg text-center py-4">Signup</h1>
+        <div className="d-flex flex-col px-8 py-4">
+          <label className="input-label-required text-sm my-2" htmlFor="email">
+            Email
+          </label>
+          <input
+            type="text"
```

Comment:
make input type as **email** and if it's a required field add required as well. The browser will display a warning state since you are using the form.